x/stakeibc/keeper/host_zone.go

```diff
@@ -158,7 +158,19 @@ func (k Keeper) IncrementValidatorSlashQueryProgress(
 		return types.ErrValidatorNotFound
 	}
 
+	// Threshold of 1% means once 1% of TVL has been breached, the interval increments
+	params := k.GetParams(ctx)
+	queryThreshold := sdk.NewDecWithPrec(int64(params.ValidatorSlashQueryThreshold), 2) // percentage
+	checkpoint := queryThreshold.Mul(sdk.NewDecFromInt(hostZone.TotalDelegations)).TruncateInt()
+
+	// Prevent division by 0 error if either the threshold or TVL is 0
+	if checkpoint.IsZero() {
+		return nil
+	}
+
 	validator.SlashQueryProgressTracker = validator.SlashQueryProgressTracker.Add(amount)
+	validator.SlashQueryInterval = validator.SlashQueryProgressTracker.Quo(checkpoint)
```

Comment:
Given the conversation offline earlier, I think this is secure but we might want to change the name to something which makes more sense in this context like previous_progress_tracker to mean this ratio which we are saving

Reply:
no longer relevant with new implementation - resolving